agent/cache-types/catalog_datacenters_test.go

```diff
@@ -53,29 +53,29 @@ func TestCatalogDatacenters(t *testing.T) {
 		})
 
 	// Fetch first time
-	result, err := typ.Fetch(cache.FetchOptions{}, &structs.DatacentersRequest{})
-	result2, err := typ.Fetch(cache.FetchOptions{LastResult: &result}, &structs.DatacentersRequest{QueryOptions: structs.QueryOptions{MustRevalidate: true}})
-	result3, err := typ.Fetch(cache.FetchOptions{LastResult: &result2}, &structs.DatacentersRequest{QueryOptions: structs.QueryOptions{MustRevalidate: true}})
+	result1, err1 := typ.Fetch(cache.FetchOptions{}, &structs.DatacentersRequest{})
+	result2, err2 := typ.Fetch(cache.FetchOptions{LastResult: &result1}, &structs.DatacentersRequest{QueryOptions: structs.QueryOptions{MustRevalidate: true}})
+	result3, err3 := typ.Fetch(cache.FetchOptions{LastResult: &result2}, &structs.DatacentersRequest{QueryOptions: structs.QueryOptions{MustRevalidate: true}})
```

Comment:
For my own understanding: Is there a reason to keep the `assert.NoError()` calls further down, or could they be moved to immediately after each of these calls? That way we could keep the variable as `err` and fail the test earlier.